.github/documentation.md

```diff
@@ -0,0 +1,202 @@
+    |   |- dependency-image-build.yml [models m1 m2]
+    |       |- [m1] build-and-push-image.yml (dep-image-c1-m1) 
+    |       |- [m2] build-and-push-image.yml (dep-image-c1-m2) 
+    |- [c2] dependency-image-pipeline.yml 
+    |   |- build-and-push-image.yml (base-spack-c2)
+    |   |- dependency-image-build.yml [models m1 m2]
+    |       |- [m1] build-and-push-image.yml (dep-image-c2-m1) 
+    |       |- [m2] build-and-push-image.yml (dep-image-c2-m2) 
+```
+
+### Model Test Pipeline
+
+This workflow seeks to build upon the Dependency Image Pipeline as explained above by taking an appropriate dependency image and attempting to install a modified (most likely from a PR) model over the top. This allows further testing and runs of a modified model without the excessive overhead of installing dependencies from scratch.
+
+#### Model Test Pipeline Overview
+
+Model repositories that implement the `model-build-test-ci.yml` starter workflow (such as the [access-nri/MOM5](https://github.com/ACCESS-NRI/MOM5/blob/master/.github/workflows/model-build-test-ci.yml) repo) will call `build-ci`s `build-package.yml` workflow.
+
+This workflow begins by inferring the 'appropriate dependency image' based on a number of factors, mostly coming from the name of the dependency image (which is of the form `build-<coupled model>-<compiler name><compiler version>-<spack_packages version>:latest`).
+
+In order to find:
+
+* `spack_packages version`: In the `setup-spack-packages` job, we take the latest tagged version of the `access-nri/spack_packages` repo.
+* `coupled model`: In the `setup-model` and `setup-build-ci` jobs, we use the name of the calling repository (eg. `cice5`) and `build-ci`s `containers/models.json` to infer the overarching `coupled model` name.
+* `compiler name`/`compiler version`: In the `setup-build-ci` job, we use all compilers from the `containers/compilers.json` file. This would mean that another matrix strategy would be in order.
+
+Given those inferences, we would be able to find the appropriate dependency images to install the modified models into.
+
+We then use those containers to upload the original `spack.lock` files (also known as lockfiles) and then attempt to install the modified model in the appropriate `spack env`. If this fails, we force a recreation of the lockfile and upload this one as well, for reference.
+
+### JSON Linter Pipeline
+
+This is a relatively simple pipeline (found in `json-lint.yml`) that looks for `*.schema.json` files in `containers`, matches them up with the associated `*.json` files and makes sure they comply with the given schema.
+
+## Reusable Workflows
+
+### build-and-push-image.yml
+
+`build-and-push-image.yml` is the most used reusable workflow. This workflow builds, caches, and pushes a given Dockerfile to a given container registry. Build args and build secrets can also be added.
+
+## Appendix
+
+### On the matrix strategy for the Dependency Image Pipeline
```

Comment:
Consider putting this in the introduction, so that there's some context to the complexity that follows.

Reply:
I think the link to it before the complex parts is sufficient, but happy to be convinced otherwise :D

Reply:
Resolved offline and moved the Appendix into the pipeline section where it is relevant